libs/ledger-live-common/src/hooks/recoverFeatureFlag.ts

```diff
@@ -94,3 +94,74 @@ export function useLoginPath(
 
   return usePath(servicesConfig, uri);
 }
+
+export function useRestore24URI(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = servicesConfig?.params?.onboardingCompleted?.restore24URI;
+  const id = servicesConfig?.params?.protectId;
+
+  return useReplacedURI(uri, id);
+}
+
+export function useRestore24Path(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = useRestore24URI(servicesConfig);
+
+  return usePath(servicesConfig, uri);
+}
+
+export function useAccountURI(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = servicesConfig?.params?.account?.homeURI;
+  const id = servicesConfig?.params?.protectId;
+
+  return useReplacedURI(uri, id);
+}
+
+export function useAccountPath(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = useAccountURI(servicesConfig);
+
+  return usePath(servicesConfig, uri);
+}
+
+export function useAlreadySeededDeviceURI(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = servicesConfig?.params?.onboardingCompleted?.alreadyDeviceSeededURI;
+  const id = servicesConfig?.params?.protectId;
+
+  return useReplacedURI(uri, id);
+}
+
+export function useAlreadySeededDevicePath(
+  servicesConfig: Feature_ProtectServicesDesktop | null,
+): string | undefined {
+  const uri = useAlreadySeededDeviceURI(servicesConfig);
+
+  return usePath(servicesConfig, uri);
+}
+
+export function useCustomPath(
+  servicesConfig: Feature_ProtectServicesDesktop | Feature_ProtectServicesMobile | null,
+  page?: string,
+  source?: string,
+  deeplinkCampaign?: string,
+): string | undefined {
+  const modelUri = usePostOnboardingURI(servicesConfig);
+  const [basicUri] = modelUri ? modelUri.split("?") : [];
+  const uri = new URL(basicUri);
+
+  if (page) uri.searchParams.append("redirectTo", page);
+  if (source) uri.searchParams.append("source", source);
+  if (source && deeplinkCampaign) {
+    uri.searchParams.append("ajs_recover_source", source);
+    uri.searchParams.append("ajs_recover_campaign", deeplinkCampaign);
+  }
```

Comment:
Not sure about the cost of creating an URL and appending to it but I would memo that part
```suggestion
  const uri = useMemo(() => {
    const [basicUri] = modelUri ? modelUri.split("?") : [];
    const uri = new URL(basicUri);

    if (page) uri.searchParams.append("redirectTo", page);
    if (source) uri.searchParams.append("source", source);
    if (source && deeplinkCampaign) {
      uri.searchParams.append("ajs_recover_source", source);
      uri.searchParams.append("ajs_recover_campaign", deeplinkCampaign);
    }
    return uri;
  }, [deeplinkCampaign, modelUri, page, source]);
```

Reply:
https://github.com/LedgerHQ/ledger-live/pull/5027/commits/843834a03d7419b1708b32efce6f1deb5cc3b701